pkg/util/dtotel/controller_runtime/client.go

```diff
@@ -0,0 +1,134 @@
+package controller_runtime
+
+import (
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/dtotel"
+	"go.opentelemetry.io/otel"
+	"golang.org/x/net/context"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ client.Client = wrappedClient{}
+
+// Reader knows how to read and list Kubernetes objects.
+type wrappedClient struct {
```

Comment:
maybe dumb question: 
instrumenting an external client, wont this cause a bunch of noise for us?
I would want the spans to "start" from code that is actually in my control.

wdyt?

Reply:
The client is doing potentially expensive I/O. So with creating spans for the client's functions, one can see if time is spent in the calling function (our code) or in the external function. It makes for better visibility of runtime.